brokers/unified/broker.go

```diff
@@ -137,42 +142,84 @@ func (b *Broker) ProcessPlugins(metas []model.PluginMeta) error {
+	}
+
+	switch strings.ToLower(meta.Type) {
+	case ChePluginType:
+		fallthrough
+	case EditorPluginType:
+		if len(meta.Spec.Extensions) != 0 {
+			return fmt.Errorf("Plugin '%s' is invalid. Field 'spec.extensions' is not allowed in plugin of type '%s'", meta.ID, meta.Type)
+		}
+		if len(meta.Spec.Containers) == 0 {
+			return fmt.Errorf("Plugin '%s' is invalid. Field 'spec.containers' must not be empty", meta.ID)
+		}
+	case TheiaPluginType:
+		fallthrough
+	case VscodePluginType:
+		if len(meta.Spec.Extensions) == 0 {
+			return fmt.Errorf("Plugin '%s' is invalid. Field 'spec.extensions' must not be empty", meta.ID)
+		}
+		if len(meta.Spec.Containers) > 1 {
+			return fmt.Errorf("Plugin '%s' is invalid. Containers list 'spec.containers' must not contain more than 1 container, but '%d' found", meta.ID, len(meta.Spec.Containers))
+		}
+		if len(meta.Spec.Endpoints) != 0 {
+			return fmt.Errorf("Plugin '%s' is invalid. Setting endpoints at 'spec.endpoints' is not allowed in plugins of type '%s'", meta.ID, meta.Type)
+		}
+	}
+	return nil
+}
+
 func validateMetas(metas []model.PluginMeta) error {
 	for _, meta := range metas {
-		switch meta.APIVersion {
-		case "":
-			return fmt.Errorf("Plugin '%s' is invalid. Field 'apiVersion' must be present", meta.ID)
-		case "v2":
-			// validate here something
-		default:
-			return fmt.Errorf("Plugin '%s' is invalid. Field 'apiVersion' contains invalid version '%s'", meta.ID, meta.APIVersion)
+		if err := ValidateMeta(meta); err != nil {
+			return err
 		}
+	}
+	return nil
+}
 
-		switch strings.ToLower(meta.Type) {
-		case ChePluginType:
-			fallthrough
-		case EditorPluginType:
-			if len(meta.Spec.Extensions) != 0 {
-				return fmt.Errorf("Plugin '%s' is invalid. Field 'spec.extensions' is not allowed in plugin of type '%s'", meta.ID, meta.Type)
-			}
-			if len(meta.Spec.Containers) == 0 {
-				return fmt.Errorf("Plugin '%s' is invalid. Field 'spec.containers' must not be empty", meta.ID)
-			}
-		case TheiaPluginType:
-			fallthrough
-		case VscodePluginType:
-			if len(meta.Spec.Extensions) == 0 {
-				return fmt.Errorf("Plugin '%s' is invalid. Field 'spec.extensions' must not be empty", meta.ID)
-			}
-			if len(meta.Spec.Containers) > 1 {
-				return fmt.Errorf("Plugin '%s' is invalid. Containers list 'spec.containers' must not contain more than 1 container, but '%d' found", meta.ID, len(meta.Spec.Containers))
-			}
-			if len(meta.Spec.Endpoints) != 0 {
-				return fmt.Errorf("Plugin '%s' is invalid. Setting endpoints at 'spec.endpoints' is not allowed in plugins of type '%s'", meta.ID, meta.Type)
-			}
+// GetPluginMeta downloads the metadata for a plugin. If specified,
+// defaultRegistry is used as the registry when plugin does not specify its registry.
+// If defaultRegistry is empty, and plugin does not specify a registry, an error is returned.
+func (b *Broker) GetPluginMeta(plugin model.PluginFQN, defaultRegistry string) (*model.PluginMeta, error) {
```

Comment:
Similarly to `ValidateMeta` above, is there a reason to export this function but keep the other internal? To me it makes more sense to export `GetPluginMetas` and keep `getPluginMeta` internal.

Reply:
An external program that would use brokers as a library could have the need to use this function on either 1 or several plugins.
Which is why the most useful public API call seemed to be the individual one. I didn't see any added value in exposing a function that simply loops on another basic API call.